### PR TITLE
fix: title and description are now correctly displayed in RepoUrlPicker

### LINKED ISSUE #20503: 🐛 Bug Report: RepoUrlPicker doesn't show its title in UI
### 📜 Description

If RepoUrlPicker is a part of parameters group then it doesn't show its title in web UI

### 👍 Expected behavior

RepoUrlPicker title should be shown in web UI

### 👎 Actual Behavior with Screenshots

**Template:**
image

**Creation of a New Component based on the template:**
image

### 👟 Reproduction steps

Add RepoUrlPicker as an item in a list of parameters in the template
Start creating the New Component based on the template
Observe that title is not shown

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

```
yarn run v1.22.19
$ /mnt/d/workspace/slb/backstage/dev-backstage/node_modules/.bin/backstage-cli info
OS:   Linux 5.10.102.1-microsoft-standard-WSL2 - linux/x64
node: v16.20.1
yarn: 1.22.19
cli:  0.22.13 (installed)
backstage:  1.17.5

Dependencies:
  @backstage/app-defaults                                          1.4.3
  @backstage/backend-app-api                                       0.4.5, 0.5.5
  @backstage/backend-common                                        0.18.5, 0.19.7
  @backstage/backend-dev-utils                                     0.1.1
  @backstage/backend-openapi-utils                                 0.0.3, 0.0.4
  @backstage/backend-plugin-api                                    0.5.4, 0.6.5
  @backstage/backend-tasks                                         0.5.10
  @backstage/catalog-client                                        1.4.4
  @backstage/catalog-model                                         1.4.2
  @backstage/cli-common                                            0.1.12
  @backstage/cli-node                                              0.1.4
  @backstage/cli                                                   0.22.13
  @backstage/config-loader                                         1.5.0
  @backstage/config                                                1.1.0
  @backstage/core-app-api                                          1.10.0
  @backstage/core-components                                       0.13.5
  @backstage/core-plugin-api                                       1.6.0
  @backstage/errors                                                1.2.2
  @backstage/eslint-plugin                                         0.1.3
  @backstage/integration-aws-node                                  0.1.6
  @backstage/integration-react                                     1.1.19
  @backstage/integration                                           1.7.0
  @backstage/plugin-api-docs                                       0.9.11
  @backstage/plugin-app-backend                                    0.3.53
  @backstage/plugin-app-node                                       0.1.5
  @backstage/plugin-auth-backend                                   0.18.8
  @backstage/plugin-auth-node                                      0.2.19, 0.3.2
  @backstage/plugin-azure-devops-backend                           0.3.29
  @backstage/plugin-azure-devops-common                            0.3.1
  @backstage/plugin-azure-devops                                   0.3.6
  @backstage/plugin-catalog-backend-module-azure                   0.1.24
  @backstage/plugin-catalog-backend-module-msgraph                 0.5.12
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.2
  @backstage/plugin-catalog-backend                                1.13.3
  @backstage/plugin-catalog-common                                 1.0.16
  @backstage/plugin-catalog-graph                                  0.2.36
  @backstage/plugin-catalog-import                                 0.9.13
  @backstage/plugin-catalog-node                                   1.4.6
  @backstage/plugin-catalog-react                                  1.8.4
  @backstage/plugin-catalog                                        1.13.0
  @backstage/plugin-events-node                                    0.2.14
  @backstage/plugin-explore-backend                                0.0.12
  @backstage/plugin-explore-common                                 0.0.1, 0.0.2
  @backstage/plugin-explore-react                                  0.0.31
  @backstage/plugin-explore                                        0.4.10
  @backstage/plugin-github-actions                                 0.6.5
  @backstage/plugin-org                                            0.6.14
  @backstage/plugin-permission-common                              0.7.8
  @backstage/plugin-permission-node                                0.7.16
  @backstage/plugin-permission-react                               0.4.15
  @backstage/plugin-proxy-backend                                  0.3.2
  @backstage/plugin-scaffolder-backend                             1.17.3
  @backstage/plugin-scaffolder-common                              1.4.1
  @backstage/plugin-scaffolder-node                                0.1.5, 0.2.5
  @backstage/plugin-scaffolder-react                               1.5.5
  @backstage/plugin-scaffolder                                     1.15.0
  @backstage/plugin-search-backend-module-catalog                  0.1.9
  @backstage/plugin-search-backend-module-explore                  0.1.9
  @backstage/plugin-search-backend-module-pg                       0.5.14
  @backstage/plugin-search-backend-module-techdocs                 0.1.9
  @backstage/plugin-search-backend-node                            1.2.9
  @backstage/plugin-search-backend                                 1.4.5
  @backstage/plugin-search-common                                  1.2.6
  @backstage/plugin-search-react                                   1.7.0
  @backstage/plugin-search                                         1.4.0
  @backstage/plugin-tech-radar                                     0.6.8
  @backstage/plugin-techdocs-backend                               1.7.2
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.0
  @backstage/plugin-techdocs-node                                  1.8.2
  @backstage/plugin-techdocs-react                                 1.1.11
  @backstage/plugin-techdocs                                       1.7.0
  @backstage/plugin-user-settings                                  0.7.10
  @backstage/plugin-vault                                          0.1.19
  @backstage/release-manifests                                     0.0.10
  @backstage/test-utils                                            1.4.3
  @backstage/theme                                                 0.4.2
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.5
Done in 6.37s.

```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by fedy97**
## Hey, I just made a Pull Request!

image

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### Related Issue

Closes #20503

### CODE REVIEW

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx [hunk 1]
```diff
@@ -157,9 +158,17 @@ export const RepoUrlPicker = (props: RepoUrlPickerProps) => {
 
   const hostType =
     (state.host && integrationApi.byHost(state.host)?.type) ?? null;
-
   return (
     <>
+      {schema.title && (
+        <Box my="8px">
```
Comment (vinzscam) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx:164]: does it look bad without the hardcoded margin? Have you tried using some responsive margin with a `number` value?

Comment (fedy97) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx]: you are right, what about this?

Comment (vinzscam) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx]: nice!

**Review by vinzscam — APPROVED** [review]
:shipit:

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx [hunk 2]
```diff
@@ -157,9 +158,17 @@ export const RepoUrlPicker = (props: RepoUrlPickerProps) => {
 
   const hostType =
     (state.host && integrationApi.byHost(state.host)?.type) ?? null;
-
   return (
     <>
+      {schema.title && (
```
Comment (acierto) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx:163]: Can you please add a unit test for it?

Comment (fedy97) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx:163]: thanks for the feedback, just added!

**Comment by benjdlambert** [comment]
I think you'll see that there's some casting required after the rebase. 

https://github.com/backstage/backstage/blob/72b7dc147f8b54f400b643182b233ab2cf651672/plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.test.tsx#L288

**Comment by fedy97** [comment]
> I think you'll see that there's some casting required after the rebase.
> 
> https://github.com/backstage/backstage/blob/72b7dc147f8b54f400b643182b233ab2cf651672/plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.test.tsx#L288

ops, thanks @benjdlambert

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**